backtrace-library/src/main/java/backtraceio/library/interfaces/Metrics.java

```diff
@@ -0,0 +1,100 @@
+import backtraceio.library.metrics.UniqueEvent;
+
+public interface Metrics {
+
+    /**
+     * Send all outgoing messages (unique and summed) currently queued
+     */
+    void send();
+
+    /**
+     * Add a unique event, such as UserID, SteamID and other attributes that uniquely identify a
+     * user. This list is persistent, meaning that events will not be removed upon Send(), as they
+     * are for summed events. For non-standard unique events, server side configuration needs to be
+     * done.
+     * Please refer to the online documentation at https://support.backtrace.io
+     * @param name          the name of the Unique event
+     * @return true if added successfully, otherwise false.
+     */
+    boolean addUniqueEvent(String name);
+
+    /**
+     * Add a unique event, such as UserID, SteamID and other attributes that uniquely identify a
+     * user. This list is persistent, meaning that events will not be removed upon Send(), as they
+     * are for summed events. For non-standard unique events, server side configuration needs to be
+     * done.
+     * Please refer to the online documentation at https://support.backtrace.io
+     * @param name          the name of the Unique event
+     * @param attributes    linked attributes which will update this unique event on update
+     * @return true if added successfully, otherwise false.
+     */
+    boolean addUniqueEvent(String name, Map<String, Object> attributes);
+
+    /**
+     * Get the pending list of unique events
+     *
+     * @return list of pending unique events to send
+     */
+    ConcurrentLinkedDeque<UniqueEvent> getUniqueEvents();
+
+    /** Adds a summed event to the outgoing queue.
+     * @see backtraceio.library.metrics.BacktraceMetrics#send()
+     * @param metricGroupName   name of the metric group to be incremented. This metric group must
+     * be configured on the server side as well.
+     * Please refer to the online documentation at https://support.backtrace.io
+     * @return true if added successfully, otherwise false.
+     */
+    boolean addSummedEvent(String metricGroupName);
+
+    /** Adds a summed event to the outgoing queue.
+     * @see backtraceio.library.metrics.BacktraceMetrics#send()
+     * @param metricGroupName   name of the metric group to be incremented. This metric group must
+     * be configured on the server side as well.
+     * Please refer to the online documentation at https://support.backtrace.io
+     * @param attributes        Custom attributes to add. Will be merged with the default
+     * attributes, with attribute values provided here overriding any defaults.
+     * @return true if added successfully, otherwise false.
+     */
+    boolean addSummedEvent(String metricGroupName, Map<String, Object> attributes);
+
+    /**
+     * Get the pending list of summed events
+     *
+     * @return list of pending summed events to send
+     */
+    ConcurrentLinkedDeque<SummedEvent> getSummedEvents();
```

Comment:
Note what the use of this is. Is it read only, can I modify, etc.

Reply:
and again probably wait a generic collection interface

Reply:
Not sure what you mean but it's already generic.

Reply:
@vlussenburg doesn't think we should be able to get and modify summed events/unique events. Is this true? I see user can get and modify events in backtrace-unity @konraddysput:
https://github.com/backtrace-labs/backtrace-unity/blob/553aab2b39c318ff96ebed4bc739bf2c87304649/Runtime/Services/BacktraceMetrics.cs#L169